source/common/protobuf/message_validator_impl.h

```diff
@@ -45,16 +55,20 @@ class WarningValidationVisitorImpl : public ValidationVisitor,
   uint64_t prestats_unknown_count_{};
 };
 
-class StrictValidationVisitorImpl : public ValidationVisitor {
+class StrictValidationVisitorImpl : public ValidationVisitor, public WipCounterBase {
 public:
-  // Envoy::ProtobufMessage::ValidationVisitor
-  void onUnknownField(absl::string_view description) override;
+  void setCounters(Stats::Counter& wip_counter) { setWipCounter(wip_counter); }
 
   // Envoy::ProtobufMessage::ValidationVisitor
+  void onUnknownField(absl::string_view description) override;
   bool skipValidation() override { return false; }
   void onDeprecatedField(absl::string_view description, bool soft_deprecation) override;
+  void onWorkInProgress(absl::string_view description) override;
 };
 
+// TODO(mattklein123): There are various places where the default strict validator is being used.
+// This does not increment the WIP stat. We should remover this as a public function and make sure
```

Comment:
Why doesn't it increment the WiP stat? You inherit from `WipCounterBase` above, is this because `setCounters` is only called on the canonical strict validator?

Reply:
Yes, will clarify.